ember-element-helper/src/helpers/element.ts

```diff
@@ -7,14 +7,24 @@ import { ensureSafeComponent } from '@embroider/util';
 
 function UNINITIALIZED() {}
 
-export default class ElementHelper extends Helper {
-  constructor() {
-    super(...arguments);
-    this.tagName = UNINITIALIZED;
-    this.componentClass = null;
-  }
+type Positional<T> = [name: T];
+type Return<T> = EmberComponent<{
+  Element: T extends keyof HTMLElementTagNameMap ? HTMLElementTagNameMap[T] : Element;
+  Blocks: { default: [] };
+}>;
+
+export interface ElementSignature<T extends string> {
```

Comment:
any reason not to use this strategy? https://github.com/tildeio/ember-element-helper/pull/102#discussion_r1186850830

Reply:
it is using those types. I stripped them away, so I can use individual parts (`Returns` and `Positional`) within the class.